### DISCUSSION TITLE
"Proxying object storage through nginx" article doesn't really work as-described

### DISCUSSION THREAD

**Discussion by mitchellwrosen** [mastodon/mastodon#20501] [Q&A]
Hey all, I'm following the official documentation, trying to set up S3 as a backend for user-uploaded files.

I've followed the instructions in https://docs.joinmastodon.org/admin/optional/object-storage-proxy/, but nginx fails to start:

```
nginx: [emerg] no "ssl_certificate" is defined for the "listen >
nginx: configuration file /etc/nginx/nginx.conf test failed
```

My main `mastodon` nginx config (created by the setup wizard) indeed has all of the necessary ssl config:

```
server {
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
  server_name mysite.com;

  ssl_protocols TLSv1.2 TLSv1.3;
  ssl_ciphers HIGH:!MEDIUM:!LOW:!aNULL:!NULL:!SHA;
  ssl_prefer_server_ciphers on;
  ssl_session_cache shared:SSL:10m;
  ssl_session_tickets off;

  # Uncomment these lines once you acquire a certificate:
  ssl_certificate     /etc/letsencrypt/live/mysite.com/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/mysite.com/privkey.pem;
  
  ...
}
```

Anyway, I'm unsure how to proceed. I see that the following step in the docs, which doesn't work due to the aforementioned error, have me actually create a cert for this subdomain:

```
certbot --nginx -d files.mysite.com
systemctl reload nginx
```

It seems to me that either the two nginx files should be merged somehow (?), or else the docs are missing a step, or perhaps they assume the reader is more familiar with nginx than I am. Thanks for any help.

**Comment by supernovae** [comment]
You should create a subdomain for your S3 storage such as images.mysite.com and then set that as your S3_ALIAS_HOST
